### ISSUE TITLE
Please use DefaultAzureCredential in CommandLine for CI/CD pipelines

### ISSUE THREAD

**Issue by Jaykul** [Particular/NServiceBus.Transport.AzureServiceBus#562]
Since we switched to managed identity, our ops team wants to use role-based assignments for everything, and are strongly opposed to any use of a SharedAccessKey.

We need to run the CommandLine in a "Azure PowerShell" or "Azure CLI" task in Azure DevOps (i.e. using a registered service connection), but it fails because it only supports connection strings.

I would like to modify https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/blob/a3f004e7c8dfcb9e96981f03ac5af793b4e47413/src/CommandLine/CommandRunner.cs#L12-L14

To be something like:
```csharp
            var connectionStringToUse = connectionString.HasValue() ? connectionString.Value() : Environment.GetEnvironmentVariable(EnvironmentVariableName);
            
            ServiceBusAdministrationClient client;
            if (connectionStringToUse.Contains("="))
            {
                client = new ServiceBusAdministrationClient(connectionStringToUse);
            }
            else
            {
                client = new ServiceBusAdministrationClient(connectionStringToUse, new DefaultAzureCredential());
            }
```

If we use the [token overload](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.administration.servicebusadministrationclient.-ctor?view=azure-dotnet#azure-messaging-servicebus-administration-servicebusadministrationclient-ctor(system-string-azure-core-tokencredential)) like that, the tool would just pick up the existing PowerShell or Az CLI authentication token (plus handling environment variables, etc.) (See [DefaultAzureCredential](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet)).

**Comment by mauroservienti** [comment]
Thanks for raising this @Jaykul. I labeled the issue as a candidate for the next release.

**Comment by danielmarbach** [comment]
@Jaykul what do you think about this approach https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/653? 

The PR would allow specifying a fully qualified namespace with `--namespace yournamespace.servicebus.windows.net` and then the default azure credential would be used.

Would that work for you?

**Comment by Jaykul** [comment]
I think that makes sense. Separating the existing `--connection-string` parameter from a new `--namespace` parameter helps with clarity. It might help people understand what's going on if you get more explicit in the help string about the intent. Maybe something like "Sets the fully qualified namespace for connecting with cached credentials, such as those from Azure PowerShell or CLI" or whatever.

**Comment by danielmarbach** [comment]
@Jaykul I used namespace as a short form for `fullyQualifiedNamespace` like the SDK uses

![image](https://user-images.githubusercontent.com/174258/198043924-c1be17ed-d4bf-40b6-b5d0-00c64ed378ce.png)

I like the suggestion

> Sets the fully qualified namespace for connecting with cached credentials, such as those from Azure PowerShell or CLI

**Comment by danielmarbach** [comment]
This will be included in the 3.1 release of the transport and command line tool which will be released shortly after 3.0 has RTMed.
